helper/openapi/openapi.go

```diff
@@ -0,0 +1,102 @@
+// Package openapi provides structures and helpers that align with a subset of version 3 of the
+// OpenAPI specification: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md
+package openapi
+
+import (
+	"github.com/hashicorp/vault/version"
+)
+
+const OpenAPIVersion = "3.0.2"
+
+type Document struct {
+	OpenAPIVersion string               `json:"openapi"`
+	Info           Info                 `json:"info"`
+	Paths          map[string]*PathItem `json:"paths"`
+}
+
+type Info struct {
+	Title   string `json:"title"`
+	Version string `json:"version"`
+}
+
+type PathItem struct {
+	Description string      `json:"description,omitempty"`
+	Parameters  []Parameter `json:"parameters,omitempty"`
+	Sudo        bool        `json:"x-vault-sudo,omitempty" mapstructure:"x-vault-sudo"`
```

Comment:
Not sure if this is in the RFC, but there are also unauthenticated paths under "special paths". I think it would be useful to highlight/mark these paths in the document as such just like we are doing with sudo.

Reply:
I agree and will add this.